src/derivative_operators/derivative_irreg_operator.jl

```diff
@@ -399,3 +399,75 @@ end
 (L::FiniteDifference)(du,u,p,t) = mul!(du,L,u)
 # get_LBC(::FiniteDifference{A,B,C,D}) where {A,B,C,D} = C
 # get_RBC(::FiniteDifference{A,B,C,D}) where {A,B,C,D} = D
+
+function LinearAlgebra.Array(A::FiniteDifference{T}) where T
+    N = A.dimension
+    L = zeros(T, N, N+2)
+    stl = A.stencil_length
+    stl_2 = div(stl,2)
+    extended_x = one(T).*[-A.dx[1]; zero(T); cumsum([A.dx; A.dx[N-1]])]
+
+    # Apply lower stencils
+    for i in 1:stl_2-1
+        L[i,1:stl] = calculate_weights(A.derivative_order, extended_x[i+1], extended_x[1:stl])
+    end
+
+    # Apply inner stencils
+    for i in max(stl_2,1):N-stl_2+1
+        L[i,i-stl_2+1:i+stl_2+1] = calculate_weights(A.derivative_order, extended_x[i+1], extended_x[i+1-stl_2:i+1+stl_2])
+    end
+
+    # Apply upper stencils
+    for i in N-stl_2+2:N
+        L[i,N-stl+3:N+2] = calculate_weights(A.derivative_order, extended_x[i+1], extended_x[N-stl+3:N+2])
+    end
+    return L
+end
+
+function SparseArrays.SparseMatrixCSC(A::FiniteDifference{T}) where T
+    N = A.dimension
+    L = spzeros(T, N, N+2)
+    stl = A.stencil_length
+    stl_2 = div(stl,2)
+    extended_x = one(T).*[-A.dx[1]; zero(T); cumsum([A.dx; A.dx[N-1]])]
+
+    # Apply lower stencils
+    for i in 1:stl_2-1
+        L[i,1:stl] = calculate_weights(A.derivative_order, extended_x[i+1], extended_x[1:stl])
+    end
+
+    # Apply inner stencils
+    for i in max(stl_2,1):N-stl_2+1
+        L[i,i-stl_2+1:i+stl_2+1] = calculate_weights(A.derivative_order, extended_x[i+1], extended_x[i+1-stl_2:i+1+stl_2])
```

Comment:
you shouldn't need to re-calculate the weights. These are calculated once and stored in the operators.

Reply:
I see that loop. My concern is that initialize_left_boundary!() and initialize_right_boundary!() for the Dirichlet, Dirichlet0, Neumann0, and Periodic boundary conditions do not create any stencilling for hte boundary points. 

I could change the implementation to use the boundary stencils computed in the Neumann, Robin, and None boundary types, however, I am unsure as how to deal with the aforementioned boundary types.

Reply:
the stencil should end up independent of the boundary, so code for what it will be and not what it currently is for now.